### PR TITLE
add condaEnvPrefix to Gvcf2hvcf and places it is called

### PR DESCRIPTION
**Pull request by lynnjo**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [x] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Added condaEnvPrefix as optional parameter to Gvcf2Hvcf.
```

### CODE REVIEW

**Review by zrm22 — APPROVED** [review]

**Review by tcasstevens — APPROVED** [review]

File: src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt [hunk 1]
```diff
@@ -493,7 +493,7 @@ class CreateMafVcf : CliktCommand(help = "Create g.vcf and h.vcf files from Anch
      * This function will bulk load sequences in from the AGC record and then will associate the returned sequences
      * with the metadata record which contains the coordiantes for the query and will add in the asmSeq.
      */
-    fun addSequencesToMetaData(dbPath: String, metadata: List<HVCFRecordMetadata>) : List<HVCFRecordMetadata> {
+    fun addSequencesToMetaData(dbPath: String, metadata: List<HVCFRecordMetadata>, condaEnvPrefix:String) : List<HVCFRecordMetadata> {
```
[the remark this answers is not in the document]
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:496]: @zrm22 The input parameter has a default of "" when it is not specified.  This is passed to each place it is used.

[the remark this answers is not in the document]
Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:496]: I would still recommend putting the default on the function header just for ease of use.

[the remark this answers is not in the document]
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:496]: This is passed through several layers, do you want it on all of the functions?:
  Gvcf2Hvcf().run() -> createASMHvcfs() -> CreateMafVcf().convertGVCFToHVCF() -> convertGVCFToHVCFForChrom() -> addSequencesToMetaData() -> retrieveAgcContigs()
 
 The latter functions are all in CreateMafVcf().  I note in CreateMafVcf() we call its version of createASMHvcfs() without passing condaEnvPrefix.  In this class, the value is taken from the input parameters.  Should this be changed to explicitly take condaEnvPrefix?  I'm thinking yes as that allows the function to be called directly from test cases or elsewhere (it is not currently called from anywhere except CreateMafVcf().run() )

[the remark this answers is not in the document]
Comment (zrm22) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:496]: I would follow it down the call stack until it gets used.  I have had to do this a number of times when working on the imputation side but it makes the functions much more flexible as they automatically can pickup the default.  

For the CreateMafVCF I would say if its used elsewhere it makes sense to have a default there as well. Just be sure to update the docs with the new optional parameter.

[the remark this answers is not in the document]
Comment (lynnjo) [reply src/main/kotlin/net/maizegenetics/phgv2/cli/CreateMafVcf.kt:496]: ok - I'll add the default throughout.  Note the optional parameter will be added to Brandon's "Convenience Commands" documentation.  He was the person who noted it was missing and prompted this additional PR to fix it.

**Review by aberthel — APPROVED** [review]

**Pull request merged**